braid/step.c

```diff
@@ -30,48 +30,142 @@ braid_Int
 _braid_Step(braid_Core         core,
             braid_Int          level,
             braid_Int          index,
+            braid_Int          calling_function,
```

Comment:
Can you explain why this new `calling_function` argument is needed?  The `ASCaller` stuff is there to give the user some context about where they are in the MGRIT solver, since user code lives outside of braid.  The delta correction implementation in internal, so I am just wondering if there is another way to do this.  We might have to talk about this in person, but feel free to give it a shot here. :)

Reply:
@cadaverous-lives @rfalgout Hi Guys, I was thinking about this too.  Is it that the calling function is needed to decide when to do certain parts of Delta Correction (internally to Braid), or does the user's Delta Correction code need calling function from Step to function properly?

Reply:
I'm not crazy about having it in the argument list of the key algorithmic routines (other than the status functions, where else is this currently used?).  However, this will require more discussion about how best to refactor everything and it's not worth the effort right now.  Since this is internal, it is easily changed later if a better approach presents itself.  Thanks!

Reply:
I took another look at _braid.h, and I see what you mean. The only functions in there now with calling_function in the arguments are _braid_Step and _braid_Residual... I'm not sure off the top of my head how I could refactor step to remove it, but I will think about it.